### PR TITLE
Editorial: Only set dtf.[[HourCycle]] once.

### PR DESCRIPTION
**Pull request by Ms2ger**
The field is set, read, and then immediately overwritten, without any other possible use in between.

### CODE REVIEW

**Review by ryzokuken — APPROVED** [review]

File: spec/datetimeformat.html [hunk 1]
```diff
@@ -95,7 +94,7 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
           1. Set _formatOptions_.[[&lt;_prop_&gt;]] to _value_.
         1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
         1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
-        1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+        1. Let _hc_ be _r_.[[hc]].
```
Comment (gibson042) [inline spec/datetimeformat.html:97]: This delocalizes references to _r_, which hinders readability for me. But I think it could work if HourCycle were addressed **before** TimeZone, e.g.
```
        …
        1. Let _r_ be ResolveLocale(%DateTimeFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %DateTimeFormat%.[[RelevantExtensionKeys]], _localeData_).
        1. Set _dateTimeFormat_.[[Locale]] to _r_.[[locale]].
        1. Set _dateTimeFormat_.[[Calendar]] to _r_.[[ca]].
        1. Set _dateTimeFormat_.[[NumberingSystem]] to _r_.[[nu]].
        1. Let _hc_ be _r_.[[hc]].
        1. Let _dataLocale_ be _r_.[[dataLocale]].
        1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
        1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
        1. If _hour12_ is *true*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, set _hc_ to *"h11"*. Otherwise, set _hc_ to *"h12"*.
        1. Else if _hour12_ is *false*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, set _hc_ to *"h23"*. Otherwise, set _hc_ to *"h24"*.
        1. Else,
          1. Assert: _hour12_ is *undefined*.
          1. If _hc_ is *null*, set _hc_ to _hcDefault_.
        1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
        1. Let _timeZone_ be ? Get(_options_, *"timeZone"*).
        …
        1. Set _dateTimeFormat_.[[TimeZone]] to _timeZone_.
        1. Let _formatOptions_ be a new Record.
        1. For each row of <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
          …
        1. Set _formatOptions_.[[hourCycle]] to _dateTimeFormat_.[[HourCycle]].
        …
```

File: spec/datetimeformat.html [hunk 2]
```diff
@@ -73,29 +73,11 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
...
-            1. Let _value_ be ? GetNumberOption(_options_, *"fractionalSecondDigits"*, 1, 3, *undefined*).
-          1. Else,
-            1. Let _value_ be ? GetOption(_options_, _prop_, *"string"*, &laquo; the strings given in the Values column of the row &raquo;, *undefined*).
-          1. Set _formatOptions_.[[&lt;_prop_&gt;]] to _value_.
         1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
         1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
-        1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+        1. Let _hc_ be _r_.[[hc]].
         1. If _hc_ is *null*, then
           1. Set _hc_ to _hcDefault_.
```
Comment (gibson042) [inline spec/datetimeformat.html:82]: This is still setting _hc_ too many times for my taste, but that can be addressed later. For the record, I'd prefer something like
```
        1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
        1. If _hour12_ is *true*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, let _hc_ be *"h11"*. Otherwise, let _hc_ be *"h12"*.
        1. Else if _hour12_ is *false*, then
          1. If _hcDefault_ is *"h11"* or *"h23"*, let _hc_ be *"h23"*. Otherwise, let _hc_ be *"h24"*.
        1. Else,
          1. Assert: _hour12_ is *undefined*.
          1. Let _hc_ be _r_.[[hc]].
          1. If _hc_ is *null*, set _hc_ to _hcDefault_.
        1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
```

File: spec/datetimeformat.html [hunk 3]
```diff
@@ -111,6 +93,23 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
...
+        1. Set _dateTimeFormat_.[[TimeZone]] to _timeZone_.
+        1. Let _formatOptions_ be a new Record.
+        1. For each row of <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
+          1. Let _prop_ be the name given in the Property column of the row.
+          1. If _prop_ is *"fractionalSecondDigits"*, then
+            1. Let _value_ be ? GetNumberOption(_options_, *"fractionalSecondDigits"*, 1, 3, *undefined*).
+          1. Else,
+            1. Let _value_ be ? GetOption(_options_, _prop_, *"string"*, &laquo; the strings given in the Values column of the row &raquo;, *undefined*).
+          1. Set _formatOptions_.[[&lt;_prop_&gt;]] to _value_.
         1. Set _formatOptions_.[[hourCycle]] to _hc_.
```
Comment (gibson042) [inline spec/datetimeformat.html:113]: Minor suggestion to avoid post hoc alias reuse:
```suggestion
        1. Set _formatOptions_.[[hourCycle]] to _dateTimeFormat_.[[HourCycle]].
```

File: spec/datetimeformat.html [hunk 4]
```diff
@@ -111,6 +93,23 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
...
           1. Assert: _hour12_ is *undefined*.
         1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
+        1. Let _timeZone_ be ? Get(_options_, *"timeZone"*).
+        1. If _timeZone_ is *undefined*, then
+          1. Let _timeZone_ be DefaultTimeZone().
+        1. Else,
+          1. Let _timeZone_ be ? ToString(_timeZone_).
+          1. If the result of IsValidTimeZoneName(_timeZone_) is *false*, then
+            1. Throw a *RangeError* exception.
+          1. Let _timeZone_ be CanonicalizeTimeZoneName(_timeZone_).
```
Comment (gibson042) [inline spec/datetimeformat.html:103]: Drive-by fixes:
```suggestion
        1. Let _timeZone_ be ? Get(_options_, *"timeZone"*).
        1. If _timeZone_ is *undefined*, then
          1. Set _timeZone_ to DefaultTimeZone().
        1. Else,
          1. Set _timeZone_ to ? ToString(_timeZone_).
          1. If the result of IsValidTimeZoneName(_timeZone_) is *false*, then
            1. Throw a *RangeError* exception.
          1. Set _timeZone_ to CanonicalizeTimeZoneName(_timeZone_).
```

**Review by gibson042 — APPROVED** [review]
A few suggestions, but none of them necessary if you disagree.

**Review by leobalter — APPROVED** [review]
